Add more license files

tillrohrmann opened the pull request:
Add LICENSE and NOTICE files for Flink modules

tillrohrmann committed:
[FLINK-10987] Add LICENSE & NOTICE files for flink-test-utils

tillrohrmann committed:
[FLINK-10987] Add LICENSE & NOTICE files for flink-shaded-hadoop2

tillrohrmann committed:
[FLINK-10987] Add LICENSE & NOTICE files for flink-shaded-hadoop2-uber

tillrohrmann committed:
[FLINK-10987] Add LICENSE & NOTICE files for flink-shaded-yarn-tests

tillrohrmann committed:
[FLINK-10987] Add LICENSE & NOTICE files for flink-shaded-curator

tillrohrmann committed:
[FLINK-10987] Do not deploy flink-yarn-tests module to repository

zentol commented on flink-shaded-hadoop/flink-shaded-yarn-tests/src/main/resources/META-INF/NOTICE:
```diff
@@ -0,0 +1,47 @@
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+This project bundles the following dependencies under the Apache Software License 2.0 (http://www.apache.org/licenses/LICENSE-2.0.txt)
```
I will have to scan over my PR again if we really want to include all ASL dependencies.

tillrohrmann replied:
We can also drop them if we agree on this. I added it because it was no additional work.

zentol replied:
Looking forward it may be better to include them; simply covering all dependencies will make it easier to detect discrepancies.

zentol commented on flink-yarn-tests/src/main/resources/META-INF/NOTICE:
```diff
@@ -0,0 +1,128 @@
+- com.thoughtworks.paranamer:paranamer:2.7
+- xmlenc:xmlenc:0.52
+
+This project bundles the following dependencies under the BSD 3-Clause license.
```
generalize to `under the BSD license.`

tillrohrmann replied:
Will do.

zentol commented on flink-shaded-hadoop/flink-shaded-hadoop2/src/main/resources/META-INF/NOTICE:
```diff
@@ -0,0 +1,31 @@
+This project bundles the following dependencies under the Apache Software License 2.0 (http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+- com.google.guava:guava:11.0.2
+- net.java.dev.jets3t:jets3t:0.9.0
```
the indentation is different to the other PRs (which use 2 spaces). It's a minor thing, but also something we can easily be consistent about.

StephanEwen commented:
I just tried to remove the `test-jar` exporting from `flink-yarn-tests` and it seems to work.
In that case, no jar is exported and we could skip all the license files in there.

That would be nice, because that module should go away soon anyways: It should merge into `flink-yarn` and only use Hadoop as a provided dependency, so re-export nothing.

tillrohrmann commented:
Thanks for the review @StephanEwen and @zentol. I've addressed your comments. We now skip the deployment of the `flink-yarn-tests` module. Due to this, we no longer need to provide a `NOTICE` file.

Merging this PR.
